umbraco/Umbraco-CMS
# Enable single block mode

### Prerequisites


If there's an existing issue for this PR then this fixes #12927

### Description
This adds Single Block Mode pretty much as described in the discussion.

I've opted for the simple boolean for enabling Single Block Mode, with the addition of a label description stating what is needed for actually enabling Single Block Mode.

When Single Block Mode is enabled, the block gets added automatically when the node is created. There is a property action for replacing the block with one from the clipboard, if there is any blocks in the clipboard.

Single Block Mode outputs a BlockListItem<> instead of the BlockListModel, so you have direct access to the block.

If you convert a "multi" Block list to a Single Block Mode Block List, it will keep the existing blocks, and let you delete all but the first one. So the editor can remove the ones not needed before publishing the page.

To test:
- Try to create a single block mode block list, by creating a block list with just one available block, min-max of 1-1, and Single Block Mode checked.
- Verify that the block is created automatically
- Copy a block, and check that you can replace the current block with the one from the clipboard.

![22-10-15---22-11-04---164---CVfaiu71Fw](https://user-images.githubusercontent.com/3726467/196006062-3d49c9e8-f642-4cff-980f-7da267fb76f6.png)
![22-10-15---22-10-35---805---iP0ortg6v2](https://user-images.githubusercontent.com/3726467/196006063-6a2c13bf-644a-4a36-b6a7-dca2b2bac61a.png)



---
_This item has been added to our backlog [AB#23582](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/23582)_
